no longer builds with make 4.3 on Linux, bad rules in Makefile

Hey @philburk ,

pforth builds on macos but not linux, using the build/unix directory.  it used to work on linux, but no more.

It seems VPATH is not being honored by gnu make 4.3 for the .c files.  This is a fresh clone of the repo:

```
(build/unix)> make all
make: *** No rule to make target 'pf_cglue.o', needed by 'pforth'.  Stop.
```

Output of make -d

[pforth_makeout.txt](https://github.com/philburk/pforth/files/5177140/pforth_makeout.txt)

--- a/build/unix/Makefile
+++ b/build/unix/Makefile
@@ -73,10 +73,10 @@ LINK = $(CC) $(LDFLAGS)
 PFOBJS     = $(PFSOURCE:.c=.o)
 PFEMBOBJS  = $(PFSOURCE:.c=.eo)
 
-.c.o: $(PFINCLUDES)
+%.o: %.c $(PFINCLUDES)
 	$(COMPILE) -c -o $@ $<
 
-.c.eo: $(PFINCLUDES) pfdicdat.h
+%.eo: %.c $(PFINCLUDES) pfdicdat.h
 	$(COMPILE) $(EMBCCOPTS) -c -o $@ $<
 
 .PHONY: all clean test

